parsing/HACKING.adoc

```diff
@@ -25,6 +25,52 @@ target of the root Makefile to rebuild the compiler parser. See
 link:../Makefile.menhir[] for the details of the various
 Menhir-related targets and their use.
 
+In addition, if your purpose is to modify the parser, it is recommended that
+you make sure to use the same version of Menhir to generate the
+updated parser than the one which has been used to produce the files
+currently stored in the `boot/menhir` subdirectory. If you need
+to change the version of Menhir used to generate the parser, then it is
+recommended that you do so in a dedicated commit which contains only
+this change and does not otherwise modify anything else.
+
+To know which version of Menhir has been used to generate the files stored
+in `boot/menhir`, you can use a command like
+
+        grep -e "require_[0-9]\+" boot/menhir/menhirLib.ml
+
+This should show a line similar to this one:
+
+        let require_20210419 = ()
+
+Here, Menhir's version is `20210419` and it is thus this version
+of Menhir you need to use to re-generate the parser, to make sure the only
+modifications introduced in `boot/menhir` are those derived from
+your changes in `parser.mly`.
```

Comment:
Note: I think it should always be okay to upgrade Menhirlib to a more recent version of Menhir, and never okay to downgrade to an older version. So the recommendation could be that if you have a more recent version of Menhir than was used in the compiler, you can have a (first, separate) commit to update MenhirLib, and then use your version.